lib/spack/spack/cmd/spec.py

```diff
@@ -49,7 +50,10 @@ def setup_parser(subparser):
     subparser.add_argument(
         '-t', '--types', action='store_true', default=False,
         help='show dependency types')
-    arguments.add_common_arguments(subparser, ['specs'])
+    subparser.add_argument(
+        'specs', nargs=argparse.REMAINDER,
+        help=('one or more package specs. when an environment is active and no '
+              'specs are provided, the environment root specs are used instead.'))
```

Comment:
I would revert this change and add an epilog:
```python
subparser.epilog = "When an environment is active and no specs are provided, the environment root specs are used instead."
```

Reply:
Thanks, didn't know that existed

Reply:
Huh, this was a commit pushed to the wrong branch :sweat_smile:

Reply:
Reverted, I'll take this as your review of #25941 then :p